### PR TITLE
Mixed datatypes - support primitive types

### PR DESCRIPTION
**Pull request by clementetb**
Implements support for primitive types on Mixed:
- Annotation processor,
- support for managed and unmanaged modes,
- and its test cases.

### CODE REVIEW

**Review by edualonso — APPROVED** [review]
I haven't thoroughly analysed the code generation part, but I assume it's correct because of the tests. It LGTM in any case. Perhaps someone else should it just in case.

File: realm/realm-annotations-processor/src/test/resources/io/realm/some_test_AllTypesRealmProxy.java [hunk 1]
```diff
@@ -1379,6 +1407,7 @@ public static String getSimpleClassName() {
                 } else {
                     objProxy.realmSet$columnDate(JsonUtils.stringToDate(reader.nextString()));
                 }
+            } else if (name.equals("columnMixed")) {
             } else if (name.equals("columnBinary")) {
```
[the remark this answers is not in the document]
Comment (edualonso) [reply realm/realm-annotations-processor/src/test/resources/io/realm/some_test_AllTypesRealmProxy.java:1411]: Oh, this is the test class for the generated code. Never mind then.

File: realm/realm-library/src/main/cpp/io_realm_internal_Table.cpp [hunk 2]
```diff
@@ -438,6 +438,335 @@ JNIEXPORT jboolean JNICALL Java_io_realm_internal_Table_nativeIsNull(JNIEnv*, jo
...
 }
 
+// ----------------- Mixed getters
+
+JNIEXPORT jint JNICALL Java_io_realm_internal_Table_nativeMixedGetType(JNIEnv* env, jclass, jlong nativeTableRefPtr,
+                                                                       jlong columnKey, jlong rowKey)
+{
+    TableRef table = TBL_REF(nativeTableRefPtr);
+    if (!TYPE_VALID(env, table, columnKey, type_Mixed)) {
+        // TODO: Throw exception
```
[the remark this answers is not in the document]
Comment (clementetb) [reply realm/realm-library/src/main/cpp/io_realm_internal_Table.cpp:448]: Forgot to replace it 😳

File: realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.kt [hunk 3]
```diff
@@ -922,11 +973,21 @@ class RealmProxyClassGenerator(private val processingEnvironment: ProcessingEnvi
                 "io.realm.MutableRealmInteger" -> {
                     emitStatement("Long %s = ((%s) object).%s().get()", getter, interfaceName, getter)
                     beginControlFlow("if (%s != null)", getter)
-                        emitStatement("Table.nativeSetLong(tableNativePtr, columnInfo.%sColKey, objKey, %s.longValue(), false)", fieldName, getter)
-                        if (isUpdate) {
-                            nextControlFlow("else")
-                                emitStatement("Table.nativeSetNull(tableNativePtr, columnInfo.%sColKey, objKey, false)", fieldName)
-                        }
+                    emitStatement("Table.nativeSetLong(tableNativePtr, columnInfo.%sColKey, objKey, %s.longValue(), false)", fieldName, getter)
```
[the remark this answers is not in the document]
Comment (clementetb) [reply realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.kt:976]: I don't think so.

**Comment by clementetb** [comment]
Merged in UUID and added support for it on Mixed

**Review by rorbech — COMMENTED** [review]
Seems like `ct/mixed-datatype-story` is not up to date with `support-new-datatypes`. That makes all the UUID changes display as new changes and makes it difficult to judge what is actually under review.

**Pull request merged**